client/src/main/java/org/apache/rocketmq/schema/registry/client/serializer/AbstractAvroSerializer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.schema.registry.client.serializer;
+
+import org.apache.avro.Schema;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.rocketmq.schema.registry.client.SchemaRegistryClient;
+import org.apache.rocketmq.schema.registry.client.exceptions.RestClientException;
+import org.apache.rocketmq.schema.registry.client.exceptions.SerializationException;
+import org.apache.rocketmq.schema.registry.common.dto.SchemaRecordDto;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.ExecutionException;
+
+public class AbstractAvroSerializer<T> {
+
+    private static final int SCHEMA_ID_LENGTH = 16;
+    protected SchemaRegistryClient schemaRegistry;
+    private final EncoderFactory encoderFactory = EncoderFactory.get();
+
+    protected byte[] serializeImpl(
+            String subject, T originMessage)
+            throws SerializationException {
+        if (schemaRegistry == null) {
+            throw new SerializationException("please initialize the schema registry client first");
+        }
+
+        if (originMessage == null) {
+            return null;
+        }
+
+        try {
+
+        } catch (Exception e) {
+            throw new SerializationException("get schema by subject failed", e);
+        }
+
+        try {
+            SchemaRecordDto schemaRecordDto = getSchemaBySubject(subject);
+            long schemaId = schemaRecordDto.getSchemaId();
```

Comment:
in our design, we use id + version to place a schema record, so you should write version in header too